tidy3d/plugins/mode/mode_solver.py

```diff
@@ -348,6 +366,421 @@ def data_raw(self) -> ModeSolverData:
+
+            def compute_sel_coord_value(_rho):
+                return _rho * sin_theta + self.rotated_bend_center[idx_v]
+
+        # Transform fields from Cartesian to cylindrical coordinates
+        for i, _rho in enumerate(rho):
+            lateral_coord_value = compute_sel_coord_value(_rho)
+            fields_at_point = {
+                field_name: getattr(mode_solver_data, field_name)
+                .sel({plane_name_normal: self.plane.center[self.normal_axis]}, method="nearest")
+                .interp({plane_name_lateral: lateral_coord_value}, method="linear")
+                .isel({plane_name_lateral: 0})
+                .values
+                for field_name in ("Ex", "Ey", "Ez", "Hx", "Hy", "Hz")
+            }
+
+            for field_type in ["E", "H"]:
+                field_data_cylindrical[field_type + "r"][i, 0] = (
+                    fields_at_point[field_type + cmp_1] * cos_theta
+                    + fields_at_point[field_type + cmp_2] * sin_theta
+                )
+                field_data_cylindrical[field_type + "theta"][i, 0] = (
+                    -fields_at_point[field_type + cmp_1] * sin_theta
+                    + fields_at_point[field_type + cmp_2] * cos_theta
+                )
+                field_data_cylindrical[field_type + "axial"][i, 0] = fields_at_point[
+                    field_type + axial_name
+                ]
+
+        coords = {
+            "rho": rho,
+            "theta": theta,
+            "axial": axial,
+            "f": f,
+            "mode_index": mode_index,
+        }
+
+        solver_data_cylindrical = {
+            name: ScalarModeFieldCylindricalDataArray(field_data_cylindrical[name], coords=coords)
+            for name in ("Er", "Etheta", "Eaxial", "Hr", "Htheta", "Haxial")
+        }
+        solver_data_cylindrical["n_complex"] = mode_solver_data.n_complex
+
+        return solver_data_cylindrical
+
+    def _mode_rotation(
+        self,
+        solver_ref_data_cylindrical: Dict[
+            Union[ScalarModeFieldCylindricalDataArray, ModeIndexDataArray]
+        ],
+        solver: ModeSolver,
+    ) -> ModeSolverData:
+        """Rotate the mode solver solution from the reference plane in cylindrical coordinates
+        to the practical plane."""
+        rotated_data_arrays = {}
+        for field_name in ("Ex", "Ey", "Ez", "Hx", "Hy", "Hz"):
+            if self.colocate is True:
+                # Get colocation coordinates in the solver plane
+                normal_dim, plane_dims = self.plane.pop_axis("xyz", self.normal_axis)
+                colocate_coords = {}
+                for dim, sym in zip(plane_dims, self.solver_symmetry):
+                    coords = self.grid_snapped.boundaries.to_dict[dim]
+                    if len(coords) > 2:
+                        if sym == 0:
+                            colocate_coords[dim] = coords[1:-1]
+                        else:
+                            colocate_coords[dim] = coords[:-1]
```

Comment:
Let's separate this out in a separate private method that we can call both here and in `_colocate_fields` to avoid code duplication.